services/ExecutorService.ts

```diff
@@ -0,0 +1,204 @@
+import { ethers } from 'ethers';
+import logger from '../utils/Logger.js';
+
+/**
+ * Service for interacting with Wormhole Executor API and generating executor parameters
+ * for L1BTCDepositorNttWithExecutor contract
+ */
+export class ExecutorService {
+  private readonly executorApiUrl: string;
+  private readonly timeout: number;
+
+  constructor(executorApiUrl: string = 'https://executor.labsapis.com/v0/quote', timeout: number = 30000) {
+    this.executorApiUrl = executorApiUrl;
+    this.timeout = timeout;
+  }
+
+  /**
+   * Generate executor parameters for a cross-chain transfer
+   * @param srcChain Source chain ID (Ethereum = 2)
+   * @param dstChain Destination chain ID (SeiEVM = 40, BaseSepolia = 10004)
+   * @param refundAddress User's address to receive refunds
+   * @param relayInstructions Gas limit and execution parameters for destination chain
+   * @returns Executor parameters including signed quote and cost
+   */
+  async generateExecutorParameters(
+    srcChain: number,
+    dstChain: number,
+    refundAddress: string,
+    relayInstructions: string = '0x01'
+  ): Promise<{
+    executorArgs: {
+      value: string;
+      refundAddress: string;
+      signedQuote: string;
+      instructions: string;
+    };
+    estimatedCost: string;
+  }> {
+    try {
+      logger.info(`Generating executor parameters for ${srcChain} -> ${dstChain}`);
+
```

Comment:
Should we guard against bad inputs here? Could be even simpler since we only support one source chain and 2 destination chains, according to paramater comments
```suggestion
  if (!VALID_CHAIN_IDS.includes(srcChain) || !VALID_CHAIN_IDS.includes(dstChain)) {
    throw new Error('Invalid chain ID provided');
  }
```